samples/tutorial/Tutorial1.md

```diff
@@ -0,0 +1,245 @@
+
+Now we have our `WelcomeWorkflow` rendering a `WelcomeScreen`, and have a layout runner that knows how to display with a `WelcomeScreen`. It's time to bind this all together and actually show it on the screen!
+
+We'll update the `TutorialActivity` to set its content using a `ViewRegistry` that points to our `LayoutRunner`'s `ViewFactory`:
+
+```kotlin
+// This doesn't look like much right now, but we'll add more layout runners shortly.
+private val viewRegistry = ViewRegistry(WelcomeLayoutRunner)
+
+class TutorialActivity : AppCompatActivity() {
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    // Use an AndroidX ViewModel to start and host an instance of the workflow runtime that runs
+    // the WelcomeWorkflow and sets the activity's content view using our view factories.
+    setContentWorkflow(
+      viewRegistry,
+      configure = { Config(WelcomeWorkflow, Unit) },
+      onResult = {}
+    )
+  }
+}
+```
+
+Now, we've created a `ViewRegistry` that consists of, so far, only our `WelcomeLayoutRunner`'s `ViewFactory`, and we're using it to set the content view of our activity. When the activity is started, it will start running the `WelcomeWorkflow`.
+
+We can finally run the app again! It will look exactly the same as before, but now it is powered by our workflow.
+
+## Driving the UI from Workflow State
+
+Right now, the workflow isn't handling any of the events from the UI. Let's update it to be responsible for the login name as well as the action when the login button is pressed.
+
+### State
+
+All workflows have a `State` type that represents the internal state of the workflow. This should be all of the data for which *this* workflow is _responsible_. It usually corresponds to the state for the UI.
+
+Let's model the first part of state that we want to track: the login `name`. Update the `State` type to include a name property. We will also need to update `initialState` to give an initial value:
+
+```kotlin
+object WelcomeWorkflow : StatefulWorkflow<Unit, State, Output, WelcomeScreen>() {
+
+  data class State(
+    val name: String
+  )
+
+  // …
+
+  override fun initialState(
+    props: Unit,
+    snapshot: Snapshot?
+  ): State = State(name = "")
+
+  // …
+}
+
+```
+
+Now that we have the state modeled, we'll send it to the UI every time a render pass happens. The text field will overwrite its value with whatever was provided.
+
+If you run the app again, you'll see that it still behaves the same, letting your type into the name field. This is because we have only rendered the screen once.
+
+You may have noticed that your workflow only has access to its `State` in a few functions, and even then in many cases it is read-only. This is intentional. The Workflow infrastructure manages the state for the various workflows that are running and triggers a re-render when appropriate. In order to update the workflow's internal state, we need to add an "Action",
```

Comment:
```suggestion
You may have noticed that your workflow only has access to its `State` in a few functions, and even then in many cases it is read-only. This is intentional. Instead of allowing your Workflow to modify the state directly, the Workflow infrastructure manages the state for the various workflows that are running and triggers a re-render when appropriate. In order to update the workflow's internal state, we need to add an "Action",
```